extensions/amp-sidebar/1.0/component.js

```diff
@@ -166,11 +167,81 @@ function SidebarWithRef(
         >
           <div className={classes.backdropOverscrollBlocker}></div>
         </div>
-      </>
-    )
+      </div>
+    </>
   );
 }
 
 const Sidebar = forwardRef(SidebarWithRef);
 Sidebar.displayName = 'Sidebar'; // Make findable for tests.
 export {Sidebar};
+
+/**
+ * @param {!SidebarDef.SidebarNavToolbar} props
+ * @return {PreactDef.Renderable}
+ */
+export function SidebarNavToolbar({
+  toolbar,
+  'toolbar-target': toolbarTarget,
+  children,
+  ...rest
+}) {
+  const ref = useRef();
+  const [targetEl, setTargetEl] = useState();
+  const [, setReRender] = useState(true);
+
+  // Trigger re-render each time the window is resized
```

Comment:
We don't actually want to re-render each time the window is resized, but each time `media.matches` changes, right? Can we use the `MediaQueryProps` class to listen?

Reply:
I'm open to using `MediaQueryProps`, but not that familiar with it.  Are you able to provide an example or elaborate a bit more?

In the current implementation, we want to render on window resize, because we check `media.matches` within the `effect` and the re-render is what triggers this `effect` to be run and check `media-matches`.